testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,165 @@
+    //
+    // For example, a developer may explicitly tag an image for repetitive
+    // testing and sharing. They may add a tag with a PR number or version.
+    //
+    // Note: the "latest" tag is automatically set by the registry, and there
+    // are no guarantees that it will point to a specific image. It should be
+    // avoided.
+    string tag = 3;
+  }
+
+  // The number of pods that should be created with this container. For example,
+  // specify 2 replicas on a client container to create 2 clients.
+  //
+  // This field will always be ignored on driver containers. It is ignored on
+  // server containers at present, but may be supported eventually.
+  int32 replicas = 4;
+}
+
+message StartTestSessionRequest {
+  // The test scenarios to run as a part of the test session.
+  repeated grpc.testing.Scenario scenarios = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The container used as the driver which runs the test scenarios.
+  //
+  // If unset, the "latest" tag is used with the standard driver.
+  Container driver_container = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  // The containers used as clients.
+  repeated Container worker_containers = 3;  // [(google.api.field_behavior) = REQUIRED];
+
+  // An optional location where the results should be written.
+  oneof result_destination {
+    // The name of the BigQuery table where results should be written.
+    string bq_results_table = 4;  // [(google.api.field_behavior) = OPTIONAL];
+
+    // The fully qualified GCS URI where a results file should be written.
+    string gcs_results_file_uri = 5;  // [(google.api.field_behavior) = OPTIONAL];
+  }
+ }
+
+message TestSessionResults {
+  // The results for each scenario.
+  repeated grpc.testing.ScenarioResult scenario_results = 1;
+
+  // The amount of time that this session lived, including scheduling to
+  // to reporting results.
+  google.protobuf.Duration time_elapsed = 2;
+}
+
+enum TestSessionPhase {
+  // The test session is waiting for idle nodes to schedule pods.
+  QUEUED = 0;
+
+  // Kubernetes is creating and starting the test pods.
+  PROVISIONING = 1;
+
+  // The driver is waiting for the workers to warmup.
+  WARMING_UP = 2;
+
+  // The tests are running and metrics are being collected.
+  TESTING = 3;
+
+  // Results are being saved to persistent storage.
+  REPORTING = 4;
+}
+
+message TestSessionMetadata {
+  // If the session has not terminated, this field provides the current phase
+  // of the test session.
+  TestSessionPhase current_phase = 1;
+
+  // The name of the kubernetes cluster where tests were run.
+  string gke_cluster = 2;
+
+  // The version of the service that is processing this request.
+  string service_version = 3;
+
+  // The time that the server created a test session. This time is not tied to
+  // any specific TestSessionPhase, so it is not indicative of a successful
+  // enqueue or provisioning.
+  google.protobuf.Timestamp create_time = 4;
```

Comment:
I thought about providing a phase history with timestamps for all the different pods that the service creates. This may be useful for debugging, but I'm not sure. Would love your thoughts?